File: Data Structures & Algorithms/remove-duplicates-from-sorted-array-ii/submission-2.kt
class Solution {
    fun removeDuplicates(nums: IntArray): Int {
        var left = 0
        var right = 0
        while (right < nums.count()) {
            var count = 1
            while (right + 1 < nums.count() && nums[right] == nums[right + 1]) {
                count += 1
                right += 1
            }
            for (i in 0 until minOf(2, count)) {
                nums[left] = nums[right]
                left += 1
            }
            right += 1
        }
        return left
    }
}

File: Data Structures & Algorithms/jump-game/submission-2.swift
class Solution {
    func canJump(_ nums: [Int]) -> Bool {

        var cache = [Int: Bool]()

        func dfs(_ i: Int) -> Bool {
            if i >= nums.count - 1 {
                return true
            }
            if cache[i] != nil {
                return cache[i]!
            }
            let step = nums[i]
            if step > 0 {
                for j in i + 1 ... i + step {
                    if dfs(j) {
                        cache[i] = true
                        return true
                    }
                }
            }
            cache[i] = false
            return false
        }

        return dfs(0)

    }
}
